### DISCUSSION TITLE
Is there a way to identify a function as an extension function and its corresponding class/type?

### DISCUSSION THREAD

**Discussion by dAnjou** [LemonAppDev/konsist#1444] [Ask a Question] [answered]
### Wanted end result.

See title.

### What you tried so far.

Searched the documentation. Browsed the API of `KoFunctionDeclaration`.

**Comment by igorwojda** [comment, marked as answer]
Yes the thing you are looking for is called `ReceiverType`

Snippet:
```kotlin
val SampleClass.sampleProperty: String
```

Konsist:
```kotlin
Konsist
    .scopeFromProject()
    .properties()
    .assertTrue {
          receiverType?.name shouldBeEqualTo "SampleClass"
    }
```

or
```kotlin
Konsist
    .scopeFromProject()
    .properties()
    .withReceiverType()
    ...
```


To make it more obvious I will add `isExtension` property https://github.com/LemonAppDev/konsist/pull/1455

BTW I will update documentation after the next release.

**Reply by wellingtoncosta** [reply]
@igorwojda is there any way to identify if the extended type (receiver type) resides in a specific module / package? When I tried to print out the receiverType.package.name, it shows the the package name where the extension function is declared, not the extended type itself.
